var/spack/repos/builtin/packages/cdo/package.py

```diff
@@ -34,13 +34,64 @@ class Cdo(Package):
     version('1.7.2', 'f08e4ce8739a4f2b63fc81a24db3ee31', url='https://code.zmaw.de/attachments/download/12760/cdo-1.7.2.tar.gz')
     version('1.6.9', 'bf0997bf20e812f35e10188a930e24e2', url='https://code.zmaw.de/attachments/download/10198/cdo-1.6.9.tar.gz')
 
-    variant('mpi', default=True)
+    variant('szip', default=True, description='Enable szip compression for GRIB1')
+    variant('hdf5', default=False, description='Enable HDF5 support')
+    variant('netcdf', default=True, description='Enable NetCDF support')
+    variant('udunits2', default=True, description='Enable UDUNITS2 support')
+    variant('grib', default=True, description='Enable GRIB_API support')
+    variant('libxml2', default=True, description='Enable libxml2 support')
+    variant('proj', default=True, description='Enable PROJ library for cartographic projections')
+    variant('curl', default=True, description='Enable curl support')
+    variant('fftw', default=True, description='Enable support for fftw3')
+    variant('magics', default=True, description='Enable Magics library support')
 
-    depends_on('netcdf')
-    depends_on('netcdf+mpi', when='+mpi')
-    depends_on('netcdf~mpi', when='~mpi')
+    depends_on('szip', when='+szip')
+    depends_on('netcdf', when='+netcdf')
+    depends_on('hdf5+threadsafe', when='+hdf5')
+    depends_on('udunits2', when='+udunits2')
+    depends_on('grib-api', when='+grib')
+    depends_on('libxml2', when='+libxml2')
+    depends_on('proj', when='+proj')
+    depends_on('curl', when='+curl')
+    depends_on('fftw', when='+fftw')
+    depends_on('magics', when='+magics') 
 
     def install(self, spec, prefix):
-        configure('--prefix={0}'.format(prefix))
+        config_args = ["--prefix=" + prefix,
+                       "--enable-shared",
+                       "--enable-static"]
+
+        if '+szip' in spec:
+            config_args.append('--with-szlib=%s' % spec['szip'].prefix)
```

Comment:
What happens when I try to build `cdo~szip` but there's a system installation of szip? I assume it will still build with szip support. You might want to add:
```python
else:
    config_args.append('--without-<package>')
```
to all of these if-statements.